### PR TITLE
The path to the checkstyle.xml is now correctly resolved on Unix

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: Tasks/Gradle/Strings/resources.resjson/en-US/resources.resjson [hunk 1]
```diff
@@ -55,11 +55,13 @@
   "loc.input.help.sqDbPassword": "SonarQube server 5.1 and lower only. Enter the password for the database user i.e. sonar.jdbc.password",
   "loc.input.label.sqAnalysisIncludeFullReport": "Include full analysis report in the build summary (SQ 5.3+)",
   "loc.input.help.sqAnalysisIncludeFullReport": "This option will delay the build until the SonarQube analysis is completed.",
+  "loc.input.label.sqAnalysisBreakBuildIfQualityGateFailed": "Fail the build on quality gate failure (SQ 5.3+)",
```
Comment (bgavrilMS) [inline Tasks/Gradle/Strings/resources.resjson/en-US/resources.resjson]: Unrelated change - the resjson should has uncommited changes

File: Tasks/Maven/Strings/resources.resjson/en-US/resources.resjson [hunk 2]
```diff
@@ -55,9 +55,11 @@
   "loc.input.help.sqDbPassword": "SonarQube server 5.1 and lower only. Enter the password for the database user i.e. sonar.jdbc.password",
   "loc.input.label.sqAnalysisIncludeFullReport": "Include full analysis report in the build summary (SQ 5.3+)",
   "loc.input.help.sqAnalysisIncludeFullReport": "This option will delay the build until the SonarQube analysis is completed.",
+  "loc.input.label.sqAnalysisBreakBuildIfQualityGateFailed": "Fail the build on quality gate failure (SQ 5.3+)",
```
Comment (bgavrilMS) [inline Tasks/Maven/Strings/resources.resjson/en-US/resources.resjson]: Unrelated change - the resjson has uncommited messages. This will lead to loc problems.

File: Tasks/Gradle/CodeAnalysis/checkstyle.gradle [hunk 3]
```diff
@@ -16,8 +16,12 @@ rootProject {
 
                 proj.apply plugin: 'checkstyle'
                 checkstyle {
-                    println(rootDir);
-                    configFile rootProject.file('checkstyle.xml')
```
Comment (bgavrilMS) [inline Tasks/Gradle/CodeAnalysis/checkstyle.gradle]: This seems to be resolved differently on windows vs linux. So much cross platform :)

**Comment by blairmc-msft** [comment]
LGTM

File: Tasks/Gradle/CodeAnalysis/checkstyle.gradle [hunk 4]
```diff
@@ -16,8 +16,12 @@ rootProject {
...
+                    // Groovy way of resolving the current directory
```
Comment (duncanpMS) [inline Tasks/Gradle/CodeAnalysis/checkstyle.gradle]: By "Groovy", I assume you're not referring to the Groovy language that runs on a JVM?
"Current directory" = "directory containing this file"?

File: Tasks/Gradle/CodeAnalysis/checkstyle.gradle [hunk 5]
```diff
@@ -16,8 +16,12 @@ rootProject {
 
                 proj.apply plugin: 'checkstyle'
                 checkstyle {
-                    println(rootDir);
-                    configFile rootProject.file('checkstyle.xml')
+                    // Groovy way of resolving the current directory  
+                    File currentDir = file(new File("."));
+                    // Usually the rules xml sits with the project, but we don't want to touch the source files
```
Comment (duncanpMS) [inline Tasks/Gradle/CodeAnalysis/checkstyle.gradle]: Some more documentation in the source would be useful e.g. _We are using a hard-coded checkstyle.xml file that is checked in with as part of this task. We expect to find it in the same folder as this file. The user cannot specify their own checkstyle.xml file._

**Comment by duncanpMS** [comment]
LGTM

**Comment by HSAR** [comment]
LGTM - thanks for committing the loc and resjson changes.
